### PR TITLE
Enable drivers for mimxrt595 evk

### PR DESCRIPTION
**Pull request by willisyi**
Enabled I2C, spi, dma, counter, wdog, flash storage for mimxrt595.

### CODE REVIEW

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33-pinctrl.dtsi
Comment (gmarull) [inline boards/arm/mimxrt595_evk/mimxrt595_evk_cm33-pinctrl.dtsi:41]: why is this needed? sensor is configuring GPIO using the GPIO API.

File: include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h
Comment (mmahadevan108) [inline include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h:2]: We can delete the `(c)` from the Copyright

Comment (ChayGuo) [reply include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h:2]: Deleted the (c), thanks.

File: samples/sensor/fxos8700/README.rst
Comment (ChayGuo) [reply samples/sensor/fxos8700/README.rst:126]: corrected, thanks.

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33-pinctrl.dtsi
Comment (ChayGuo) [reply boards/arm/mimxrt595_evk/mimxrt595_evk_cm33-pinctrl.dtsi:41]: It's indeed not needed. Removed it. Thanks.

File: tests/drivers/spi/spi_loopback/boards/mimxrt595_evk_cm33.conf
Comment (ChayGuo) [reply tests/drivers/spi/spi_loopback/boards/mimxrt595_evk_cm33.conf]: Yes, tested with DMA. I've rebased the two commits for SPI and DMA enablement and split them to 1 for DMA support another for SPI support. 
Thanks.

File: boards/arm/mimxrt595_evk/Kconfig.defconfig
Comment (ChayGuo) [reply boards/arm/mimxrt595_evk/Kconfig.defconfig:28]: Accepted, thanks.

File: soc/arm/nxp_imx/rt5xx/soc.h
Comment (ChayGuo) [reply soc/arm/nxp_imx/rt5xx/soc.h:23]: > e may not be needed. Please delete.

Removed. Thanks.

**Comment by dleach02** [comment]
@hakehuang can you test suite on this PR?

**Comment by hakehuang** [comment]
> @hakehuang can you test suite on this PR?

@dleach02 , test is running will feedback once ready

**Review by danieldegrasse — DISMISSED** [review]

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts
Comment (gmarull) [inline boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts:116]: note: this should be moved to soc dts files

File: include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h
Comment (gmarull) [inline include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h:26]: nit: align using tabs as other entries

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.yaml
Comment (gmarull) [inline boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.yaml:26]: where is this tag used?

**Review by gmarull — COMMENTED** [review]
lgtm, a few minor comments

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.yaml
Comment (ChayGuo) [reply boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.yaml:26]: Deleted it since it's not used currently.

File: include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h
Comment (ChayGuo) [reply include/zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h:26]: Fixed. Thanks. 

> nit: align using tabs as other entries

File: boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts
Comment (ChayGuo) [reply boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts:116]: If I'm understanding correctly, the #address-cells, #size-cells properties are used for the child node "fxos8700@1e" which is defined below. So they are put here not in soc dts. 
Correct me if I'm wrong. 

Thanks. 
> note: this should be moved to soc dts files

**Comment by ChayGuo** [comment]
Updated the PR per review comments and resolved the conflicts in wdt_basic_api/testcase.yaml

**Review by nvlsianpu — COMMENTED** [review]
flash driver changes looks fine to me.
Needs approval from platform maintainer,

File: boards/arm/mimxrt595_evk/doc/index.rst
Comment (mmahadevan108) [inline boards/arm/mimxrt595_evk/doc/index.rst:87]: Add FLASH to this list

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c
Comment (ChayGuo) [reply drivers/flash/flash_mcux_flexspi_mx25um51345g.c:45]: > Can we avoid this platform specific define and use the same CMD for other platforms that use this flash device. Also is there a benefit to using `STR` over `DTR`

For RT685 EVK, the Octal flash is connected to Flexspi PortB, and no DQS pin is available, so if DTR used, the flexspi clock is restricted to <50MHz slower than STR(can be 100MHz), and RT685 ROM uses STR mode. 
For RT595 EVK the Octal flash uses DQS pin and can make higher performance using DTR mode, the ROM also uses DTR mode. 
This is platform/board-specific.

File: boards/arm/mimxrt595_evk/doc/index.rst
Comment (ChayGuo) [reply boards/arm/mimxrt595_evk/doc/index.rst:87]: > Add FLASH to this list

Added. Thanks.

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c
Comment (mmahadevan108) [reply drivers/flash/flash_mcux_flexspi_mx25um51345g.c:45]: Then should this be a `Kconfig` that is enabled at the board level.

Comment (ChayGuo) [reply drivers/flash/flash_mcux_flexspi_mx25um51345g.c:45]: > Then should this be a `Kconfig` that is enabled at the board level.

Thanks, added a Kconfig choice in the flash kconfig file, and the value can be set at board level.

**Review by mmahadevan108 — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**
